plugins/main/bridge/bridge.go

```diff
@@ -485,41 +493,44 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	if err := ipam.ExecDel(n.IPAM.Type, args.StdinData); err != nil {
-		return err
-	}
-
 	if args.Netns == "" {
 		return nil
 	}
 
-	// There is a netns so try to clean up. Delete can be called multiple times
-	// so don't return an error if the device is already removed.
-	// If the device isn't there then don't try to clean up IP masq either.
-	var ipnets []*net.IPNet
-	err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
-		var err error
-		ipnets, err = ip.DelLinkByNameAddr(args.IfName)
-		if err != nil && err == ip.ErrLinkNotFound {
-			return nil
+	isLayer3 := n.IPAM.Type != ""
+
+	if isLayer3 {
+		if err := ipam.ExecDel(n.IPAM.Type, args.StdinData); err != nil {
```

Comment:
This line needs to stay before `if args.Netns == "" {` which is on line 496 as of my review.

Reason is that we want to clean up ipam even when the namespace has disappeared.